test/mongo_util.py

```diff
@@ -1,23 +1,106 @@
 
-MONGO_EXE_PATH="MONGO_EXE_PATH"
-MONGO_TEMP_DIR="MONGO_TEMP_DIR"
-
-
-def get_mongo_info() -> Tuple[Path, Path]:
+# Mongo config
+TEST_CONFIG_PATH = "HANDLE_SERVICE_TEST_CFG"
+TEST_SECTION = "HandleService2"
+TEST_MONGO_EXE = "test.mongo.exe"
+TEST_TEMP_DIR = "test.temp.dir"
+TEST_USE_WIRED_TIGER = "test.mongo.wired_tiger"
+TEST_DELETE_TEMP_DIR = "test.delete.temp.dir"
+
+# Deploy config
+TEST_AUTH_SERVICE_URL = "auth-service-url"
+TEST_AUTH_URL = "auth-url"
+TEST_SHOCK_URL = "shock-url"
+TEST_ADMIN_TOKEN = "admin-token"
+TEST_ADMIN_ROLES = "admin-roles"
+TEST_NAME_SPACE = "namespace"
+TEST_DATABASE_NAME = "mongo-database"
+TEST_SCRATCH = "scratch"
+
+
+def get_config() -> Tuple[List[Union[Path, bool]], dict[str, str]]:
     """
-    Returns a tuple of
-    * The path to the mongo executable from the environment
-    * the path to a root directory for temporary mongo data from the environment.
+    Returns:
+        Mongo config that stores mongo executable, temporary directory,
+            wired_tiger, and delete_temp_dir
+        Deploy config that stores auther_serice_url, auth_url, shock_url,
+            admin_token, admin_roles, namespace, db_name, and scratch
     """
-    return (Path(os.environ.get(MONGO_EXE_PATH)), Path(os.environ.get(MONGO_TEMP_DIR)))
+    config_path = _get_config_file_path()
+    section = _get_test_config(config_path)
+
+    mongo_exe_path = _get_value(section, TEST_MONGO_EXE, config_path, True)
+    mongo_temp_dir = _get_value(section, TEST_TEMP_DIR, config_path, True)
+    wired_tiger = _get_value(section, TEST_USE_WIRED_TIGER, config_path, False)
+    delete_temp_dir = _get_value(section, TEST_DELETE_TEMP_DIR, config_path, False)
+
+    auth_serivce_url = _get_value(section, TEST_AUTH_SERVICE_URL, config_path, True)
+    auth_url = _get_value(section, TEST_AUTH_URL, config_path, True)
+    shock_url = _get_value(section, TEST_SHOCK_URL, config_path, True)
+    admin_token = _get_value(section, TEST_ADMIN_TOKEN, config_path, True)
+    admin_roles = _get_value(section, TEST_ADMIN_ROLES, config_path, True)
+    name_space = _get_value(section, TEST_NAME_SPACE, config_path, True)
+    db_name = _get_value(section, TEST_DATABASE_NAME, config_path, True)
+    scratch = _get_value(section, TEST_SCRATCH, config_path, True)
+
+    mongo_config = [
+        Path(mongo_exe_path),
+        Path(mongo_temp_dir),
+        wired_tiger == "true",
+        delete_temp_dir != "false",
+    ]
+
+    deploy_config = {
+        TEST_AUTH_SERVICE_URL: auth_serivce_url,
+        TEST_AUTH_URL: auth_url,
+        TEST_SHOCK_URL: shock_url,
+        TEST_ADMIN_TOKEN: admin_token,
+        TEST_ADMIN_ROLES: admin_roles,
+        TEST_NAME_SPACE: name_space,
+        TEST_DATABASE_NAME: db_name,
+        TEST_SCRATCH: scratch,
+    }
+
+    return mongo_config, deploy_config
+
+
+def _get_config_file_path() -> str:
+    config_path = os.environ.get(TEST_CONFIG_PATH)
+    if not config_path:
+        raise ValueError(
```

Comment:
Codecov is complaining this line isn't covered by tests, but it shouldn't be - you might need to configure it differently to not look at test code for coverage

Reply:
👍

Reply:
:+1: